File: two_pointers/11_container_with_most_water.java
package two_pointers;

// https://leetcode.com/problems/container-with-most-water/

class Solution {
    public int maxArea(int[] height) {
        
        if (height == null || height.length <= 1)
            return 0;
        
        int currVolumn = 0, maxVolumn = 0;
        int left = 0, right = height.length - 1;
        while (left < right){
            currVolumn = Math.min(height[left], height[right]) * (right - left);
            maxVolumn = Math.max(maxVolumn, currVolumn);
            if (height[left] < height[right]) left++;
            else right--;
        }

        return maxVolumn;
    }
}

File: two_pointers/11_container_with_most_water.py
# https://leetcode.com/problems/container-with-most-water/

from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:

        currVolumn, maxVolumn = 0, 0
        left, right = 0, len(height) - 1

        while left < right:
            currVolumn = min(height[left], height[right]) * (right - left)
            maxVolumn = max(maxVolumn, currVolumn)

            if height[left] < height[right]: left += 1
            else: right -= 1

        return maxVolumn
